File: server/src/main/java/com/example/server/service/AuthService.java
package com.example.server.service;

import com.example.server.dto.AuthRequest;
import com.example.server.dto.AuthResponse;
import com.example.server.dto.UserRequest;
import jakarta.servlet.http.HttpServletResponse;

public interface AuthService {
    AuthResponse login(AuthRequest request, HttpServletResponse response);
    
    AuthResponse register(UserRequest request, HttpServletResponse response);
    
    AuthResponse refreshToken(String refreshToken, HttpServletResponse response);
    
    void logout(String refreshToken, HttpServletResponse response);
} 

File: client/src/lib/services/auth.service.ts
import { apiClient } from "../api/client";

export interface LoginRequest {
  email: string;
  password: string;
}

export interface RegisterRequest {
  email: string;
  password: string;
  firstName: string;
  lastName: string;
  roles: string[];
}

export interface AuthResponse {
  accessToken: string;
  roles: number[];
}

export interface RefreshTokenRequest {
  refreshToken: string;
}

export const authService = {
  async login(data: LoginRequest): Promise<AuthResponse> {
    const response = await apiClient.post<AuthResponse>("/auth/login", data);

    return response.data;
  },

  async register(data: RegisterRequest): Promise<AuthResponse> {
    const response = await apiClient.post<AuthResponse>("/auth/registration", data);

    return response.data;
  },

  async refreshToken(data: RefreshTokenRequest): Promise<AuthResponse> {
    const response = await apiClient.post<AuthResponse>("/auth/refresh", data);

    return response.data;
  },

  async logout(refreshToken: string): Promise<void> {
    await apiClient.post("/auth/logout", { refreshToken });
  },
};
